dart/test/http_client/sentry_http_client_test.dart

```diff
@@ -199,7 +168,49 @@ void main() {
       expect(mockHub.captureExceptionCalls.length, 0);
       verify(mockClient.close());
     });
+
+    test('Breadcrumb has correct duration', () async {
+      final sut = Fixture.getSut(client: MockClient((request) async {
+        expect(request.url, requestUri);
+        await Future.delayed(Duration(seconds: 1));
+        return Response('', 200, reasonPhrase: 'OK');
+      }));
+
+      final response = await sut.client.get(requestUri);
+      expect(response.statusCode, 200);
+
+      expect(sut.hub.addBreadcrumbCalls.length, 1);
+      final breadcrumb = sut.hub.addBreadcrumbCalls.first.crumb;
+
+      var durationString = breadcrumb.data!['duration']! as String;
+      // we don't check for anything below a second
+      expect(durationString.startsWith('0:00:01'), true);
+    });
   });
 }
 
 class CloseableMockClient extends Mock implements BaseClient {}
+
+class Fixture {
+  static Fixture getSut({MockClient? client}) {
+    return Fixture._(mockClient: client);
+  }
+
+  Fixture._({MockClient? mockClient}) {
+    hub = MockHub();
+
+    final mc = mockClient ??
+        MockClient((request) async {
+          expect(request.url, requestUri);
+          return Response('', response, reasonPhrase: reasonPhrase);
+        });
+
+    client = SentryHttpClient(client: mc, hub: hub);
+  }
+
+  late MockHub hub;
+  late SentryHttpClient client;
+
+  int response = 200;
+  String? reasonPhrase;
+}
```

Comment:
this is still not following the fixture pattern, it should not have a static method, let me refactor for you

Reply:
```
// only a single instance of Fixture
// void main() {
//  final fixture = Fixture();
//}
class Fixture {
  
  // every test does final sut = fixture.getSut() either passing MockClient or not
  SentryHttpClient getSut({MockClient? mockClient) {
        final mc = mockClient ?? getClient();
        return SentryHttpClient(client: mc, hub: hub);
  }

  // tests could assert or expect fixture.hub
  late MockHub hub = MockHub();

  // could optionally have reasonPhrase, status etc with defaults values
  // so tests could also do sut = fixture.getSut(fixture.getClient(500, 'my reason'))
  MockClient getClient() {
    MockClient((request) async {
          expect(request.url, requestUri);
          return Response('', 200, reasonPhrase: null);
        });
  }
}
```

Reply:
guess you can have a better idea now, let me know if its not clear

Reply:
Thanks 😅